src/features/authentication/profile/profileAction.js

```diff
@@ -0,0 +1,178 @@
+import toastr from 'toastr';
+import {
+  IS_LOADING, IS_COMPLETE,
+  PROFILE_VIEW, PROFILE_UPDATE, PROFILE_UPDATE_USER_IMAGE,
+} from '../../../shared/constants/ActionTypes';
+import fetchData from '../../../shared/utilities/fetchData';
+import history from '../../../shared/utilities/history';
+import localStorageUtil from '../../../shared/utilities/localStorageUtil';
+
+export const viewUserProfile = profile => ({
+  type: PROFILE_VIEW,
+  profile,
+});
+
+export const updateUserProfile = profile => ({
+  type: PROFILE_UPDATE,
+  profile,
+});
+
+export const updateUserImage = image => ({
+  type: PROFILE_UPDATE_USER_IMAGE,
+  image,
+});
+
+
+/**
+ * @description action to View User Profile Picture
+ * @param {object} Users profile - contains users profile
+ * @returns {object} success/failed users profile
+ */
+export const getUserProfile = userProfile => async (dispatch, getState) => {
+  const state = getState();
+  const { token, username } = state.auth.user;
+  try {
+    dispatch({ type: IS_LOADING });
+    const response = await fetchData({
+      url: `users/${username}`,
+      method: 'get',
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': `${token}`,
+      },
+      data: userProfile,
+    });
+    dispatch({ type: IS_COMPLETE });
+    if (response.statusText === 'Unauthorized') {
+      return history.push('/login');
+    } if (response.status === 200) {
+      localStorageUtil.setItem('ah_user', Object.assign({}, { ...state.auth.user }, { ...response.data.user }));
```

Comment:
@Oluwafayokemi, I see that you are using object spread syntax and Object.assign together. Both of them are actually doing the same thing. You may want to either use Object.assign or use the spread syntax which is a more succinct syntax for it.